Integrations/URLHaus/URLHaus.py

```diff
@@ -0,0 +1,493 @@
+
+''' GLOBALS/PARAMS '''
+
+# Remove trailing slash to prevent wrong URL path to service
+API_URL = demisto.params()['url'].rstrip('/')
+
+# Should we use SSL
+USE_SSL = not demisto.params().get('insecure', False)
+
+# Remove proxy if not set to true in params
+if not demisto.params().get('proxy'):
+    os.environ.pop('HTTP_PROXY', None)
+    os.environ.pop('HTTPS_PROXY', None)
+    os.environ.pop('http_proxy', None)
+    os.environ.pop('https_proxy', None)
+
+THRESHOLD = int(demisto.params().get('threshold', 1))
+
+COMPROMISED_IS_MALICIOUS = demisto.params().get('compromised_is_malicious', False)
+
+# Headers to be sent in requests
+HEADERS = {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Accept': 'application/json'
+}
+
+''' HELPER FUNCTIONS '''
+
+
+def http_request(method, command, data=None):
+    url = f'{API_URL}/{command}/'
+    demisto.info(f'{method} {url}')
+    res = requests.request(method,
+                           url,
+                           verify=USE_SSL,
+                           data=data,
+                           headers=HEADERS)
+
+    if res.status_code != 200:
+        raise Exception(f'Error in API call {url} [{res.status_code}] - {res.reason}')
+
+    return res
+
+
+def reformat_date(date):
+    try:
+        return dt.strptime(date.rstrip(' UTC'), '%Y-%m-%d %H:%M:%S').strftime('%Y-%m-%dT%H:%M:%S')
+    except Exception:
+        return 'Unknown'
+
+
+def extract_zipped_buffer(buffer):
+    with io.BytesIO() as bio:
+        bio.write(buffer)
+        with zipfile.ZipFile(bio) as z:
+            return z.read(z.namelist()[0])
+
+
+def query_url_information(url):
+    return http_request('POST',
+                        'url',
+                        f'url={url}')
+
+
+def query_host_information(host):
+    return http_request('POST',
+                        'host',
+                        f'host={host}')
+
+
+def query_payload_information(hash_type, hash):
+    return http_request('POST',
+                        'payload',
+                        f'{hash_type}_hash={hash}')
+
+
+def download_malware_sample(sha256):
+    return http_request('GET',
+                        f'download/{sha256}')
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+def test_module():
+    """
+    Performs basic get request to get item samples
+    """
+    http_request('POST', 'url')
+
+
+def calculate_dbot_score(blacklists, threshold, compromised_is_malicious):
+    blacklist_appearances = []
+    for blacklist, status in blacklists.items():
+        if blacklist == 'spamhaus_dbl':
+            if status.endswith('domain') or (status.startswith('abused') and compromised_is_malicious):
+                blacklist_appearances.append((blacklist, status))
+        elif status == 'listed':
+            blacklist_appearances.append((blacklist, None))
+
+    if len(blacklist_appearances) >= threshold:
+        description = ''
+        for appearance in blacklist_appearances:
+            if not appearance[1]:
```

Comment:
What if `appearance[1]` is `0`,`''`? Might be safer to use `if appearance[1] is None:`